include/sys/zio.h

```diff
@@ -131,7 +131,7 @@ enum zio_encrypt {
  * The number of "legacy" compression functions which can be set on individual
  * objects.
  */
-#define	ZIO_COMPRESS_LEGACY_FUNCTIONS ZIO_COMPRESS_LZ4
+#define	ZIO_COMPRESS_LEGACY_FUNCTIONS ZIO_COMPRESS_ZSTD
```

Comment:
This should not be changed here, this is used for send stream compatibility:

        /*
         * Compression function must be legacy, or explicitly enabled.
         */
        if ((BP_GET_COMPRESS(bp) >= ZIO_COMPRESS_LEGACY_FUNCTIONS &&
            !(dsp->dsa_featureflags & DMU_BACKUP_FEATURE_LZ4)))
                return (B_FALSE);

Reply:
mmh i unsterstand what it means. i just dont understand why it is used as assert check in dmu_object_set_compress. any compression level higher or equal the legacy_functions may raise a assert there. so this code i mentioned can be wrong then. what do you think?